lib/Target/x86_64/x86_64GOT.h

```diff
@@ -98,18 +98,44 @@ class x86_64GOTPLT0 : public x86_64GOT {
   mutable uint8_t Value[24];
 };
 
+/** \class x86_64GOTPLTN
+ *  \brief GOT entry for PLT function (lazy binding).
+ *
+ *  Each function with a PLT entry has a corresponding GOTPLTN entry.
+ *  Initially points to PLT entry + 6 (the pushq instruction).
+ *  Updated by dynamic linker on first call to point to actual function.
+ */
 class x86_64GOTPLTN : public x86_64GOT {
 public:
   x86_64GOTPLTN(ELFSection *O, ResolveInfo *R)
-      : x86_64GOT(GOT::GOTPLTN, O, R, /*Align=*/8, /*Size=*/8) {}
+      : x86_64GOT(GOT::GOTPLTN, O, R, /*Align=*/8, /*Size=*/8),
+        m_PLTEntry(nullptr) {}
 
   x86_64GOT *getFirst() override { return this; }
 
   x86_64GOT *getNext() override { return nullptr; }
 
-  static x86_64GOTPLTN *Create(ELFSection *O, ResolveInfo *R) {
-    return make<x86_64GOTPLTN>(O, R);
+  // Link to corresponding PLT entry for address calculation
+  void setPLTEntry(Fragment *plt) { m_PLTEntry = plt; }
+
+  virtual llvm::ArrayRef<uint8_t> getContent() const override {
+    Value = 0;
+    ELFSection *pltSection =
+        m_PLTEntry->getOwningSection()->getOutputELFSection();
+    if (pltSection) {
+      // Point to PLT entry + 6 (pushq instruction)
+      Value = pltSection->addr() + m_PLTEntry->getOffset() + 6;
+    }
+
+    return llvm::ArrayRef(reinterpret_cast<const uint8_t *>(&Value),
```

Comment:
Do you need to cast it to `const`? Ideally, `reinterpret_cast` should not be used for changing constness.